### DISCUSSION TITLE
How do I patch a dependency's dependency ?

### DISCUSSION THREAD

**Discussion by cassepipe** [pnpm/pnpm#6418] [Q&A]
I am using `vite` and I'd like to apply that [patch](https://github.com/http-party/node-http-proxy/pull/1559/files) to its `http-proxy` dependency.

I have used the `pnpm patch` and ` pnpm patch-commit` commint successfully on a dependency installed by me  but can I do it on a dependency's dependency ? If yes, how ?

Thanks in advance.

**Comment by cassepipe** [comment]
I have tried to put myself in "nodes_module/vite" and patch there but I dont' think this is supposed to work because it doesn't want to patch where there isn't a lock file

**Reply by kristiannotari** [reply]
I think you can still patch a package that you don't specify as a direct dependency, if that somehow get to your node modules due to it being a dependency of one of your dependencies. Isn't just patching it directly working?

**Reply by cassepipe** [reply]
I have installed the `http-proxy` dependency directly and it turns out that the latest version is compatible with the dependency listed by `vite` so I am assuming that yes in this case, installing and patching that dependancy directly solves this.
What if it wasn't the case though and `vite` used a older imcompatible version though ?

**Reply by kristiannotari** [reply]
I meant: couldn't you patch it WITHOUT installing it directly? If you install it directly you then just install the compatible version, the same vite is depending on, no problems there from what I can see.

**Reply by cassepipe** [reply]
I *can't* with the `pnpm patch` command since pnpm is not able to find the package in `node_modules` if it's not installed directly.
Maybe there's a way but I don't know it.

I am sorry if it feels like I am not understanding something obvious. Please enlighthen if you feel I need to be.

**Reply by kristiannotari** [reply]
no no, I'm no wizard when it comes to pnpm. I just thought you could patch dependencies of dependencies, cause I'm doing that thing in one of my repo. I'm patching an eslint dependency which I do no declare directly. But I have a monorepo setup and that dependency is a dependency of one of my dev dependencies in the root package.json, so maybe it's treated differently cause it ends up inside node_modules and I can patch it directly
